xla/service/gpu/ir_emitter_unnested.cc

```diff
@@ -967,6 +968,17 @@ absl::Status IrEmitterUnnested::EmitFusedMHAThunk(
       TF_ASSIGN_OR_RETURN(bias_slice, GetAllocationSliceForHlo(bias));
       bias_shape = bias->shape();
     }
+
+    bool has_seqlen_qk =
```

Comment:
Maybe rewrite `has_seqlen_qk` as:

```
bool has_seqlen_qk = seqlen_qk_operand_index == instr->operand_count() - 3;
```

I think that should be equivalent, because:

```
2 == (instr->operand_count() - 3 - has_mask - has_bias); <-> 2 + has_mask + has_bias == intr->operand_count() - 3
```

Given this rewrite, I am wondering what the last operand of `instr` is? It seems `seqlen_qk_operand_index` is for q, `selen_qk_operand_index + 1` is for k.

Reply:
Also, in case this is a bug, and it should in fact be checking

```
has_seqlen_qk = seqlen_qk_operand_index == instr->operand_count() - 2
```

then, add a test that exposes this bug.

Reply:
Fixed. operand order should be Q, K, V, mask*, bias*, seqlen_q*, seqlen_k* with last 4 being optional. Not sure how to test this without cudnn graph generation(will be in a separate PR). The new cuDNN front-end PR is still pending https://github.com/openxla/xla/pull/9778.

Reply:
Thanks for the fix, and I trust that you will add a test once it is possible.

Reply:
Will do. Thanks for the quick approval.